### PR TITLE
tidb-kubernetes: add backup and restore usage docs

### PR DESCRIPTION
**Pull request by onlymellb**
### What is changed, added or deleted? 
Add new backup and restore usage documents, the new version of the tidb-operator that supports this feature has not been released yet, so currently put these documents under dev directory first, and then move to the official directory after releasing the new version of the tidb-operator.

### What is the related PR or file link(s)? 

* [backup-gcs](https://github.com/onlymellb/docs-cn/blob/add-backup-usage-docs/dev/tidb-in-kubernetes/backup/backup-gcs.md)
* [backup-s3](https://github.com/onlymellb/docs-cn/blob/add-backup-usage-docs/dev/tidb-in-kubernetes/backup/backup-s3.md)
* [restore](https://github.com/onlymellb/docs-cn/blob/add-backup-usage-docs/dev/tidb-in-kubernetes/backup/restore.md)
### Which version does your change affect? 

dev

### CODE REVIEW

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
+---
+title: Kubernetes 上备份 TiDB 集群到 GCS
+category: how-to
+---
+
+# Kubernetes 上备份 TiDB 集群到 GCS
+这篇文档详细描述了如何将 Kubernetes 上的 TiDB 集群数据备份到 [gcs](https://cloud.google.com/storage/docs/) 上, 这里说到的备份均是指全量备份(定时备份和 Ad-hoc 备份), 底层通过使用 [`mydumper`](/dev/reference/tools/mydumper.md) 获取集群的逻辑备份，然后在将备份数据上传到远端 gcs
```
Comment (DanielZhangQD) [inline dev/tidb-in-kubernetes/backup/backup-gcs.md]: Add link to `定时备份和 Ad-hoc 备份`?

Comment (DanielZhangQD) [reply dev/tidb-in-kubernetes/backup/backup-gcs.md]: BTW, suggest moving the current doc for backup and restore to this directory.

File: dev/tidb-in-kubernetes/backup/backup-s3.md [hunk 2]
```diff
@@ -0,0 +1,310 @@
+---
```
Comment (DanielZhangQD) [inline dev/tidb-in-kubernetes/backup/backup-s3.md]: Most of the comments for gcs.md are also applicable to this doc.

File: dev/tidb-in-kubernetes/backup/restore.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+title: Kubernetes 上从备份数据恢复 TiDB 集群
+category: how-to
+---
+
+# Kubernetes 上从备份数据恢复 TiDB 集群
+这篇文档详细描述了如何将 Kubernetes 上通过 tidb-operator 备份的 TiDB 集群数据恢复的具体操作过程, 底层通过使用 [`loader`](/dev/reference/tools/loader.md) 来进行集群恢复
+
+## Ad-hoc 全量备份
+
+Ad-hoc 全量备份用过创建一个自定义的 `Backup` CR 对象来描述一次备份，tidb-operator 根据这个 `Backup` 对象来完成具体的备份过程，如果备份过程中出现错误程序不会自动重试，此时需要人工介入处理。
```
Comment (DanielZhangQD) [inline dev/tidb-in-kubernetes/backup/restore.md]: `Backup` -> `Restore`?

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
+---
+title: Kubernetes 上备份 TiDB 集群到 GCS
+category: how-to
+---
```
Comment (DanielZhangQD) [inline dev/tidb-in-kubernetes/backup/backup-gcs.md]: Many `s3` in this doc.

File: dev/tidb-in-kubernetes/backup/restore.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    ```
+
+2. 创建 `backup-demo1-tidb-secret` secret, 里面存放用来访问 TiDB 集群的 root 账号和密钥
+
+ kubectl create secret generic backup-demo1-tidb-secret --from-literal=user=root --from-literal=password=12345678 --namespace=test1
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl create secret generic restore-demo2-tidb-secret --from-literal=user=root --from-literal=password=xxx --namespace=test2
```
Comment (DanielZhangQD) [inline dev/tidb-in-kubernetes/backup/restore.md]: `backup` -> `restore`

File: dev/tidb-in-kubernetes/backup/restore.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+     ```shell
+     kubectl get bk -n test1
+     ```
+
+`.spec.secretName`：访问备份恢复目标 TiDB 集群所需密码的 secret
+
+`.spec.storageClassName`：备份恢复时需要指定使用的 PV 类型，如果不指定则默认使用 tidb-operator 启动参数中 `default-backup-storage-class-name` 指定的值，这个值默认为 `standard`
+
+`.spec.storageSize`：备份时指定备份使用的 PV 大小，这个值要大于备份恢复 TiDB 集群数据的大小
```
Comment (DanielZhangQD) [inline dev/tidb-in-kubernetes/backup/restore.md]: `备份` -> `恢复`？

File: dev/tidb-in-kubernetes/backup/backup-gcs.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (onlymellb) [reply dev/tidb-in-kubernetes/backup/backup-gcs.md]: Now do nothing, I will add cleaning logic later

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+title: Kubernetes 上备份 TiDB 集群到 GCS
```
[the remark this answers is not in the document]
Comment (onlymellb) [reply dev/tidb-in-kubernetes/backup/backup-gcs.md]: Which directory?

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+category: how-to
+---
+
+# Kubernetes 上备份 TiDB 集群到 GCS
+这篇文档详细描述了如何将 Kubernetes 上的 TiDB 集群数据备份到 [gcs](https://cloud.google.com/storage/docs/) 上, 这里说到的备份均是指全量备份(定时备份和 Ad-hoc 备份), 底层通过使用 [`mydumper`](/dev/reference/tools/mydumper.md) 获取集群的逻辑备份，然后在将备份数据上传到远端 gcs
+
+## Ad-hoc 全量备份
+
+Ad-hoc 全量备份用过创建一个自定义的 `Backup` CR 对象来描述一次备份，tidb-operator 根据这个 `Backup` 对象来完成具体的备份过程，如果备份过程中出现错误程序不会自动重试，此时需要人工介入处理。
```
[the remark this answers is not in the document]
Comment (onlymellb) [reply dev/tidb-in-kubernetes/backup/backup-gcs.md]: This is simple enough, I don't think it's necessary to make a chart.

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+
+`.spec.tidbSecretName`: 访问 TiDB 集群所需密码的 secret
+
+`.spec.storageClassName`: 备份时需要指定使用的 PV 类型，如果不指定则默认使用 tidb-operator 启动参数中 `default-backup-storage-class-name` 指定的值，这个值默认为 `standard`
+
+`.spec.storageSize`: 备份时指定需要使用的 PV 大小，这个值要大于备份 TiDB 集群数据的大小
+
+## 定时全量备份
+
+通过用户设置的备份策略定时备份 TiDB 集群，同时可配置备份的保留策略，定时全量备份通过自定义的 `BackupSchedule` CR 对象来描述，每次到备份时间点会触发一次全量备份，定时全量备份底层是通过 Ad-hoc 全量备份来实现。下面是创建定期全量备份的具体步骤：
```
Comment (shuijing198799) [inline dev/tidb-in-kubernetes/backup/backup-gcs.md]: 通过用户设置 改为 用户通过设置 ？ 同时可配置备份的保留策略能达到什么目标 ？

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+
+ ```shell
+ kubectl get bk -l tidb.pingcap.com/backup-schedule=demo1-backup-schedule-gcs -n test1
+ ```
+
+从上面的例子中我们可以看出 `backupSchedule` 的配置主要由两部分组成，一部分是 backupSchedule 独有的配置，还有一部分就是 backupTemplate，backupTemplate 指定 gcs 存储相关的配置，这块的配置和 Ad-hoc 全量备份到 gcs 的配置完全一样，因此这里我们不对这块进行过多的解释。下面主要针对 backupSchedule 独有的一些配置项进行详细介绍：
+
+ `.spec.maxBackups`: 一种备份保留策略，定时备份最多保留的备份个数，超过这个数目，就会将过时的备份删除，如果将这个最大备份保留格式设置为0，则表示保留所有备份。
+
+ `.spec.maxReservedTime`：这个和上面的参数一样，也是种一种备份保留策略，这个是按时间保留，比如设置为 `24h`, 代表只保留最近 24 小时内的备份条目，超过这个时间的备份都会被清除。如果最大备份保留个数和最大备份保留时间两者都被设置了，则以最大备份保留时间为准。
```
Comment (shuijing198799) [inline dev/tidb-in-kubernetes/backup/backup-gcs.md]: 列一下 time.praseduration 的网址 ？https://golang.org/pkg/time/#ParseDuration

File: dev/tidb-in-kubernetes/backup/backup-s3.md [hunk 11]
```diff
@@ -0,0 +1,269 @@
...
+category: how-to
+---
+
+# Kubernetes 上备份 TiDB 集群到 S3 兼容后端存储
+
+这篇文档详细描述了如何将 Kubernetes 上的 TiDB 集群数据备份到 S3 兼容的存储上, 这里说到的备份均是指全量备份(定时全量备份和 Ad-hoc 全量备份), 底层通过使用 [`mydumper`](/dev/reference/tools/mydumper.md) 获取集群的逻辑备份，然后在将备份数据上传到远端 S3 兼容的存储上。
+
+## Ad-hoc 全量备份
+
+Ad-hoc 全量备份用过创建一个自定义的 `Backup` CR 对象来描述一次备份，tidb-operator 根据这个 `Backup` 对象来完成具体的备份过程，如果备份过程中出现错误程序不会自动重试，此时需要人工介入处理。目前 S3 兼容的存储中我们只对 `Ceph`、`Amazon S3` 这两种存储做过测试，因此下面我们主要针对 `Ceph` 和 `Amazon S3` 这两种存储的使用来说明。理论上来说其余 S3 兼容的存储也可以正常的工作。
```
Comment (shuijing198799) [inline dev/tidb-in-kubernetes/backup/backup-s3.md]: 用过 改为 通过

File: dev/tidb-in-kubernetes/backup/backup-s3.md [hunk 12]
```diff
@@ -0,0 +1,269 @@
...
+* dreamhost(Dreamhost DreamObjects)
+* ibmcos(IBM COS S3)
+* minio(Minio Object Storage)
+* netease(Netease Object Storage (NOS))
+* wasabi(Wasabi Object Storage)
+* other(Any other S3 compatible provider)
+
+## 定时全量备份
+
+通过用户设置的备份策略定时备份 TiDB 集群，同时可配置备份的保留策略，定时全量备份通过自定义的 `BackupSchedule` CR 对象来描述，每次到备份时间点会触发一次全量备份，定时全量备份底层是通过 Ad-hoc 全量备份来实现。下面是创建定期全量备份的具体步骤：
```
Comment (shuijing198799) [inline dev/tidb-in-kubernetes/backup/backup-s3.md]: 用户通过设置

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 13]
```diff
@@ -0,0 +1,204 @@
...
+
+gcs 支持的 storageClass 类型有如下几种：
+
+* MULTI_REGIONAL
+* REGIONAL
+* NEARLINE
+* COLDLINE
+* DURABLE_REDUCED_AVAILABILITY
+
+如果不设置，默认使用 `COLDLINE`, 这几种存储类型的详细介绍参考 gcs [官方文档](https://cloud.google.com/storage/docs/storage-classes)。
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-gcs.md]:
```suggestion
如果不设置 `storageClass`，则默认使用 `COLDLINE`。这几种存储类型的详细介绍可参考 [GCS 官方文档](https://cloud.google.com/storage/docs/storage-classes)。
```

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 14]
```diff
@@ -0,0 +1,204 @@
...
+
+gcs 支持的 object ACL 策略有如下几种：
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-gcs.md]:
```suggestion
GCS 支持以下几种 object ACL 策略：
```

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 15]
```diff
@@ -0,0 +1,204 @@
...
+
+* authenticatedRead
+* bucketOwnerFullControl
+* bucketOwnerRead
+* private
+* projectPrivate
+* publicRead
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-gcs.md]: Please frame each one of these items with `backquote`.

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 16]
```diff
@@ -0,0 +1,204 @@
...
+
+如果不设置，默认策略为 `private`, 这几种访问控制策略的详细介绍参考 gcs [官方文档](https://cloud.google.com/storage/docs/access-control/lists)。
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-gcs.md]:
```suggestion
如果不设置 object ACL 策略，则默认使用 `private` 策略。这几种访问控制策略的详细介绍可参考 [GCS 官方文档](https://cloud.google.com/storage/docs/access-control/lists)。
```

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 17]
```diff
@@ -0,0 +1,204 @@
...
+
+gcs 支持的 bucket ACL 策略有如下几种：
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-gcs.md]:
```suggestion
GCS 支持以下几种 bucket ACL 策略：
```

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 18]
```diff
@@ -0,0 +1,204 @@
...
+
+* authenticatedRead
+* private
+* projectPrivate
+* publicRead
+* publicReadWrite
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-gcs.md]: Please frame each one of these items with `backquote`.

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 19]
```diff
@@ -0,0 +1,204 @@
...
+
+如果不设置，默认策略为 `private`, 这几种访问控制策略的详细介绍参考 gcs [官方文档](https://cloud.google.com/storage/docs/access-control/lists)。
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-gcs.md]:
```suggestion
如果不设置 bucket ACL 策略，则默认策略为 `private`。这几种访问控制策略的详细介绍可参考 [GCS 官方文档](https://cloud.google.com/storage/docs/access-control/lists)。
```

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 20]
```diff
@@ -0,0 +1,204 @@
...
+category: how-to
+---
+
+# Kubernetes 上备份 TiDB 集群到 GCS
+
+这篇文档详细描述了如何将 Kubernetes 上的 TiDB 集群数据备份到 [gcs](https://cloud.google.com/storage/docs/) 上, 这里说到的备份均是指全量备份(定时全量备份和 Ad-hoc 全量备份), 底层通过使用 [`mydumper`](/dev/reference/tools/mydumper.md) 获取集群的逻辑备份，然后在将备份数据上传到远端 gcs。
+
+## Ad-hoc 全量备份
+
+Ad-hoc 全量备份通过创建一个自定义的 `Backup` CR 对象来描述一次备份，tidb-operator 根据这个 `Backup` 对象来完成具体的备份过程，如果备份过程中出现错误程序不会自动重试，此时需要人工介入处理。
```
[the remark this answers is not in the document]
Comment (onlymellb) [reply dev/tidb-in-kubernetes/backup/backup-gcs.md]: This `Backup` object uses a declarative way to describe a backup behavior.  The program launches a backup operation based on the declaration of this `Backup` object.  `CR` means [Custom Resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/).

File: dev/tidb-in-kubernetes/backup/restore.md [hunk 21]
```diff
@@ -0,0 +1,87 @@
...
+
+`.spec.backup`： 备份源的一个 backup CR 名字，我们可以通过下面的命令获取备份集群下面有哪些备份条目，然后从中选择一个要恢复的 backup 进行恢复即可
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl get bk -n test1
+```
+
+`.spec.secretName`：访问备份恢复目标 TiDB 集群所需访问凭证的 secret
```
[the remark this answers is not in the document]
Comment (onlymellb) [reply dev/tidb-in-kubernetes/backup/restore.md]: 备份恢复，是指恢复目标

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 22]
```diff
@@ -0,0 +1,204 @@
...
+category: how-to
+---
+
+# 在 Kubernetes 上备份 TiDB 集群到 GCS
+
+本文档详细描述了如何将 Kubernetes 上 TiDB 集群的数据备份到 [Google Cloud Storage (GCS)](https://cloud.google.com/storage/docs/) 上。本文档中的“备份”，均是指全量备份（Ad-hoc 全量备份和定时全量备份），底层通过使用 [`mydumper`](/dev/reference/tools/mydumper.md) 获取集群的逻辑备份，然后再将备份数据上传到远端 GCS。
+
+## Ad-hoc 全量备份
+
+Ad-hoc 全量备份通过创建一个自定义的 `Backup` CR 对象来描述一次备份。TiDB Operator 根据这个 `Backup` 对象来完成具体的备份过程。如果备份过程中出现错误，程序不会自动重试，此时需要手动处理。
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-gcs.md]:
```suggestion
Ad-hoc 全量备份通过创建一个自定义的 `Backup` custom resource (CR) 对象来描述一次备份。TiDB Operator 根据这个 `Backup` 对象来完成具体的备份过程。如果备份过程中出现错误，程序不会自动重试，此时需要手动处理。
```

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 23]
```diff
@@ -0,0 +1,204 @@
...
+
+为了更好地描述备份的使用方式，本文档提供如下备份示例。示例假设对部署在 Kubernetes `test1` 这个 namespace 中的 TiDB 集群 `demo1` 进行数据备份，下面是具体操作过程：
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-gcs.md]:
```suggestion
为了更好地描述备份的使用方式，本文档提供如下备份示例。示例假设对部署在 Kubernetes `test1` 这个 namespace 中的 TiDB 集群 `demo1` 进行数据备份，下面是具体操作过程。
```

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 24]
```diff
@@ -0,0 +1,204 @@
...
+
+1. 在 `test1` 这个 namespace 中创建备份需要的 RBAC 相关资源。下载文件 [backup-rbac.yaml](https://github.com/pingcap/tidb-operator/blob/master/manifests/backup/backup-rbac.yaml)：
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl apply -f backup-rbac.yaml -n test1
+    ```
+
+2. 创建 `gcs-secret` secret。该 secret 存放用于访问 GCS 的凭证。`google-credentials.json` 文件存放用户从 GCP console 上下载的 service account key。具体操作参考 [GCP 官方文档](https://cloud.google.com/docs/authentication/getting-started)：
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-gcs.md]:
```suggestion
2. 创建 `gcs-secret` secret。该 secret 存放用于访问 GCS 的凭证。`google-credentials.json` 文件存放用户从 GCP console 上下载的 service account key。具体操作参考 [GCP 官方文档](https://cloud.google.com/docs/authentication/getting-started)。
```

File: dev/tidb-in-kubernetes/backup/backup-gcs.md [hunk 25]
```diff
@@ -0,0 +1,204 @@
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl create secret generic gcs-secret --from-file=credentials=./google-credentials.json -n test1
+    ```
+
+3. 创建 `backup-demo1-tidb-secret` secret。该 secret 存放用于访问 TiDB 集群的 root 账号和密钥：
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-gcs.md]:
```suggestion
3. 创建 `backup-demo1-tidb-secret` secret。该 secret 存放用于访问 TiDB 集群的 root 账号和密钥。
```

File: dev/tidb-in-kubernetes/backup/restore.md [hunk 26]
```diff
@@ -0,0 +1,87 @@
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl create secret generic restore-demo2-tidb-secret --from-literal=user=root --from-literal=password=<password> --namespace=test2
+    ```
+
+## 将指定备份恢复到 TiDB 集群
+
+1. 创建 restore CR，将指定的备份数据恢复至 TiDB 集群：
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/restore.md]:
```suggestion
1. 创建 restore custom resource (CR)，将指定的备份数据恢复至 TiDB 集群：
```

File: dev/tidb-in-kubernetes/backup/restore.md [hunk 27]
```diff
@@ -0,0 +1,87 @@
...
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl get bk -n test1
+```
+
+`.spec.secretName`：访问需恢复的目标 TiDB 集群所需凭证的 secret。
+
+`.spec.storageClassName`：备份恢复时需要指定使用的 PV 类型。如果不指定该项，则默认使用 TiDB Operator 启动参数中 `default-backup-storage-class-name` 所指定的值，这个值默认为 `standard`。
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/restore.md]:
```suggestion
`.spec.storageClassName`：恢复备份数据时需要指定使用的 PV 类型。如果不指定该项，则默认使用 TiDB Operator 启动参数中 `default-backup-storage-class-name` 所指定的值，这个值默认为 `standard`。
```

File: dev/tidb-in-kubernetes/backup/restore.md [hunk 28]
```diff
@@ -0,0 +1,87 @@
...
+
+`.spec.storageSize`：恢复集群时指定所需的 PV 大小。这个值应大于备份恢复 TiDB 集群数据的大小。
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/restore.md]: `备份恢复`是`备份和恢复`吗？

File: dev/tidb-in-kubernetes/backup/restore.md [hunk 29]
```diff
@@ -0,0 +1,87 @@
...
+     kubectl get rt -n test2 -owide
+     ```
+
+restore CR 各个字段的详细解释：
+
+`.spec.metadata.namespace`： 需要恢复的目标 TiDB 集群所在的 namespace。
+
+`.spec.cluster`：需要恢复的目标 TiDB 集群的名字。
+
+`.spec.backupNamespace`：备份集群所在的 namespace。因为当前的 restore 所需的远端存储的访问信息是通过源备份信息获取的，所以进行恢复时需要知道备份源来自哪个 namespace。
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/restore.md]: 这里的 `当前 restore` 是什么意思呢？
```suggestion
`.spec.backupNamespace`：备份集群所在的 namespace。因为当前的 restore 所需的远端存储的访问信息是通过源备份信息获取的，所以进行恢复时需要知道备份源来自哪个 namespace。
```

File: dev/tidb-in-kubernetes/backup/backup-s3.md [hunk 30]
```diff
@@ -0,0 +1,269 @@
+---
+title: Kubernetes 上备份 TiDB 集群到 S3 兼容后端存储
+category: how-to
+---
+
+# Kubernetes 上备份 TiDB 集群到 S3 兼容后端存储
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-s3.md]: 是 `S3 兼容后端存储` 还是 `兼容 S3 的后端存储`呢？

File: dev/tidb-in-kubernetes/backup/restore.md (hunk 29 above)
Comment (onlymellb) [reply dev/tidb-in-kubernetes/backup/restore.md]: 指目前的实现

File: dev/tidb-in-kubernetes/backup/restore.md (hunk 28 above)
Comment (onlymellb) [reply dev/tidb-in-kubernetes/backup/restore.md]: 我改成备份集群的数据大小

File: dev/tidb-in-kubernetes/backup/backup-s3.md [hunk 31]
```diff
@@ -0,0 +1,267 @@
...
+        secretName: s3-secret
+        endpoint: http://10.0.0.1:30074
+      storageType: s3
+      cluster: demo1
+      tidbSecretName: backup-demo1-tidb-secret
+      storageClassName: local-storage
+      storageSize: 10Gi
+    ```
+
+上面两个例子中分别将 TiDB 集群的数据全量导出备份到 Amazon S3 和 ceph，Amazon S3 的 `region`、`acl`、`enpoint`、`storageClass` 均可以省略，其余非 Amazon S3 的但是兼容 S3 的存储均可使用和 Amazon S3 类似的配置，不需要配置的字段跳过即可，参考上面例子中 ceph 的配置。
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-s3.md]:
```suggestion
以上两个示例分别将 TiDB 集群的数据全量导出备份到 Amazon S3 和 Ceph 上。Amazon S3 的 `region`、`acl`、`enpoint`、`storageClass` 配置项均可以省略。其余非 Amazon S3 的但是兼容 S3 的存储均可使用和 Amazon S3 类似的配置。可参考上面例子中 Ceph 的配置，省略不需要配置的字段。
```

File: dev/tidb-in-kubernetes/backup/backup-s3.md [hunk 32]
```diff
@@ -0,0 +1,267 @@
...
+
+Amazon S3 支持的 ACL 策略有如下几种：
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-s3.md]:
```suggestion
Amazon S3 支持以下几种 ACL 策略：
```

File: dev/tidb-in-kubernetes/backup/backup-s3.md [hunk 33]
```diff
@@ -0,0 +1,267 @@
...
+* `private`
+* `public-read`
+* `public-read-write`
+* `authenticated-read`
+* `bucket-owner-read`
+* `bucket-owner-full-control`
+
+如果不设置 ACL 策略，则默认使用 `private` 策略。这几种访问控制策略的详细介绍参考 AWS [官方文档](https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html)。
+
+Amazon S3 支持的 storageClass 类型有如下几种：
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-s3.md]: 同 Line 107 的 comment。

File: dev/tidb-in-kubernetes/backup/backup-s3.md [hunk 34]
```diff
@@ -0,0 +1,267 @@
...
+
+`.spec.metadata.namespace`: 备份 TiDB 集群所在的 namespace。
+
+`.spec.storageType`: 代表备份的存储类型，目前主要有 S3 和 GCS。
+
+`.spec.cluster`: 备份 TiDB 集群的名字。
+
+`.spec.tidbSecretName`: 访问 TiDB 集群所需密码的 secret。
+
+`.spec.storageClassName`: 备份时需的 PV 类型。如果不指定该项，则默认使用 TiDB Operator 启动参数中 `default-backup-storage-class-name` 指定的值，这个值默认为 `standard`。
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-s3.md]:
```suggestion
`.spec.storageClassName`: 备份时所需的 PV 类型。如果不指定该项，则默认使用 TiDB Operator 启动参数中 `default-backup-storage-class-name` 指定的值，这个值默认为 `standard`。
```

File: dev/tidb-in-kubernetes/backup/backup-s3.md [hunk 35]
```diff
@@ -0,0 +1,267 @@
...
+
+`.spec.storageSize`: 备份时指定所需的 PV 大小。这个值要大于备份 TiDB 集群的数据大小。
+
+更多支持的兼容 S3 的 provider 如下：
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-s3.md]: `provider` 改成`供应商`？

File: dev/tidb-in-kubernetes/backup/backup-s3.md [hunk 36]
```diff
@@ -0,0 +1,267 @@
...
+
+* `alibaba`(Alibaba Cloud Object Storage System (OSS) formerly Aliyun)
+* `digitalocean`(Digital Ocean Spaces)
+* `dreamhost`(Dreamhost DreamObjects)
+* `ibmcos`(IBM COS S3)
+* `minio`(Minio Object Storage)
+* `netease`(Netease Object Storage (NOS))
+* `wasabi`(Wasabi Object Storage)
+* `other`(Any other S3 compatible provider)
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-s3.md]: 这些供应商名是否不需要反引号？只有配置项、参数名或行内代码才需要反引号包裹。

File: dev/tidb-in-kubernetes/backup/backup-s3.md [hunk 37]
```diff
@@ -0,0 +1,267 @@
...
+          # acl: private
+          # endpoint:
+        storageType: s3
+        cluster: demo1
+        tidbSecretName: backup-demo1-tidb-secret
+        storageClassName: local-storage
+        storageSize: 10Gi
+    ```
+
+2. 创建 backupSchedule CR 开启 TiDB 集群的定时全量备份，将数据备份到 ceph。
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-s3.md]: 文档中同时存在 backupSchedule CR 与 `backupSchedule` CR，`Backup` CR 与 backup CR，请使用统一的风格。 
```suggestion
2. 创建 backupSchedule CR 开启 TiDB 集群的定时全量备份，将数据备份到 Ceph。
```

File: dev/tidb-in-kubernetes/backup/backup-s3.md [hunk 38]
```diff
@@ -0,0 +1,267 @@
...
+
+```shell
+kubectl get bk -l tidb.pingcap.com/backup-schedule=demo1-backup-schedule-s3 -n test1
+```
+
+从以上两个示例可知，`backupSchedule` 的配置由两部分组成。一部分是 `backupSchedule` 独有的配置，另一部分是 `backupTemplate`。`backupTemplate` 指定 S3 兼容存储相关的配置，该配置与 Ad-hoc 全量备份到兼容 S3 的存储配置完全一样，可参考[备份数据到兼容 S3 的存储](#备份数据到兼容-s3-的存储)。下面介绍 `backupSchedule` 独有的配置项：
+
+`.spec.maxBackups`：一种备份保留策略，决定定时备份最多可保留的备份个数。超过该数目，就会将过时的备份删除。如果将该项设置为 `0`，则表示保留所有备份。
+
+`.spec.maxReservedTime`：一种备份保留策略，按时间保留备份。比如将该参数设置为 `24h`，表示只保留最近 24 小时内的备份条目。超过这个时间的备份都会被清除。时间设置格式参考[`func ParseDuration`](https://golang.org/pkg/time/#ParseDuration)。如果同时设置最大备份保留个数和最长备份保留时间，则以最长备份保留时间为准。
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/backup/backup-s3.md]:
```suggestion
`.spec.maxReservedTime`：一种备份保留策略，按时间保留备份。例如将该参数设置为 `24h`，表示只保留最近 24 小时内的备份条目。超过这个时间的备份都会被清除。时间设置格式参考 [`func ParseDuration`](https://golang.org/pkg/time/#ParseDuration)。如果同时设置最大备份保留个数和最长备份保留时间，则以最长备份保留时间为准。
```

File: dev/tidb-in-kubernetes/backup/backup-s3.md (hunk 35 above)
Comment (onlymellb) [reply dev/tidb-in-kubernetes/backup/backup-s3.md]: 这个算是技术术语，我觉得不需要翻译

Comment (TomShawn) [reply dev/tidb-in-kubernetes/backup/backup-s3.md]: OK

**Comment by lilin90** [comment]
Currently, these three new files only apply to the latest development version of TiDB Operator and will be added to TOC later.

**Comment by DanielZhangQD** [comment]
@onlymellb Need to update the docs since https://github.com/pingcap/tidb-operator/pull/1276 merged.

**Pull request merged**